Index: src/mod/server.mod/servmsg.c
===================================================================
RCS file: /usr/local/cvsroot/eggdrop1.8/src/mod/server.mod/servmsg.c,v
retrieving revision 1.6
diff -u -r1.6 servmsg.c
--- src/mod/server.mod/servmsg.c	15 Jun 2012 00:04:59 -0000	1.6
+++ src/mod/server.mod/servmsg.c	4 Jan 2013 19:17:15 -0000
@@ -1338,6 +1338,7 @@
   strcpy(botname, origbotname);
   /* Start alternate nicks from the beginning */
   altnick_char = 0;
+  check_tcl_event("preinit-server");
   if (pass[0])
     dprintf(DP_MODE, "PASS %s\n", pass);
   dprintf(DP_MODE, "NICK %s\n", botname);
