Index: src/mod/server.mod/server.c
===================================================================
RCS file: /usr/local/cvsroot/eggdrop1.8/src/mod/server.mod/server.c,v
retrieving revision 1.8
diff -u -r1.8 server.c
--- src/mod/server.mod/server.c	25 Oct 2011 16:47:20 -0000	1.8
+++ src/mod/server.mod/server.c	4 Jan 2013 19:17:14 -0000
@@ -46,6 +46,8 @@
 static int flud_time;           /* msg flood time */
 static int flud_ctcp_thr;       /* ctcp flood threshold */
 static int flud_ctcp_time;      /* ctcp flood time */
+static char preinitserver[121]; /* what, if anything, to send to the server
+                                   immediately on connection */
 static char initserver[121];    /* what, if anything, to send to the
                                  * server on connection */
 static char botuserhost[121];   /* bot's user@host (refreshed whenever the
@@ -1356,6 +1358,7 @@
   {"altnick",             altnick,        NICKMAX,           0},
   {"realname",            botrealname,    80,                0},
   {"init-server",         initserver,     120,               0},
+  {"preinit-server",      preinitserver,  120,               0},
   {"connect-server",      connectserver,  120,               0},
   {"stackable-commands",  stackablecmds,  510,               0},
   {"stackable2-commands", stackable2cmds, 510,               0},
Index: src/mod/server.mod/servmsg.c
===================================================================
RCS file: /usr/local/cvsroot/eggdrop1.8/src/mod/server.mod/servmsg.c,v
retrieving revision 1.6
diff -u -r1.6 servmsg.c
--- src/mod/server.mod/servmsg.c	15 Jun 2012 00:04:59 -0000	1.6
+++ src/mod/server.mod/servmsg.c	4 Jan 2013 19:17:15 -0000
@@ -1338,6 +1338,9 @@
   strcpy(botname, origbotname);
   /* Start alternate nicks from the beginning */
   altnick_char = 0;
+  if (preinitserver[0])
+    do_tcl("preinit-server", preinitserver); /* Call Tcl preinit-server */
+  check_tcl_event("preinit-server");
   if (pass[0])
     dprintf(DP_MODE, "PASS %s\n", pass);
   dprintf(DP_MODE, "NICK %s\n", botname);
Index: src/mod/server.mod/help/set/server.help
===================================================================
RCS file: /usr/local/cvsroot/eggdrop1.8/src/mod/server.mod/help/set/server.help,v
retrieving revision 1.4
diff -u -r1.4 server.help
--- src/mod/server.mod/help/set/server.help	1 Nov 2010 22:38:34 -0000	1.4
+++ src/mod/server.mod/help/set/server.help	4 Jan 2013 19:17:15 -0000
@@ -20,6 +20,12 @@
    specifies text to send to the server when first connecting
    there.  this is usually used to set the bot's mode to +i
    or something equally unimportant.
+%{help=set preinit-server}%{+n}
+###  %bset preinit-server%b <raw-irc-text>
+   specifies text to send to the server as soon as the TCP
+   connection is established.
+
+see also: set init-server
 %{help=set botnick}%{+n}
 ###  %bset botnick%b
    this read-only variable returns the bots current nickname on irc.
